### PR TITLE
Kinto/remove players on disconnect

### PR DESCRIPTION
**Pull request by kyleai**
- added logic to remove players on disconnect

small changes:
- modified logic for player connecting so client has access to UUID
- added enum for events to eliminate hardcoding strings
- client recieves other players locations on connection

### CODE REVIEW

File: src/client/scenes/mainScene.ts [hunk 1]
```diff
@@ -55,10 +53,14 @@ export default class MainScene extends Phaser.Scene implements UpdateListener {
       frameRate: 10,
       repeat: -1
     });
+    this.socket = new Socket(this)
```
Comment (azadebrahimian) [inline src/client/scenes/mainScene.ts:56]: Add an extra empty line in between

File: src/client/scenes/mainScene.ts [hunk 2]
```diff
@@ -67,45 +69,73 @@ export default class MainScene extends Phaser.Scene implements UpdateListener {
...
+    this.players.get(id).destroy()
+    this.players.delete(id)
+  }
 
-    if (sprite.x != this.prev_location[0] || sprite.y != this.prev_location[1]) {
-      this.socket.updatePos(sprite.x, sprite.y)
-      this.prev_location = [sprite.x, sprite.y]
+  update() {
+    let cursors = this.input.keyboard.createCursorKeys()
+    if (this.controlling_id) {
```
Comment (azadebrahimian) [inline src/client/scenes/mainScene.ts:79]: Where is the closing curly bracket for this opening one?

File: src/client/scenes/mainScene.ts [hunk 3]
```diff
@@ -67,45 +69,73 @@ export default class MainScene extends Phaser.Scene implements UpdateListener {
...
 
   // callbacks
-
   updateLocation(player: string, x: integer, y: integer) {
     console.log("update position of " + player)
     console.log(this.players)
+    if (player == this.controlling_id) {
+      return
+    }
     if (!this.players.has(player)) {
```
Comment (azadebrahimian) [inline src/client/scenes/mainScene.ts:111]: Empty line

File: src/client/scenes/mainScene.ts [hunk 4]
```diff
@@ -67,45 +69,73 @@ export default class MainScene extends Phaser.Scene implements UpdateListener {
...
-      this.newPlayer(player)
+      this.createPlayer(player)
+    }
+    let dx = x - this.players.get(player).x
```
Comment (azadebrahimian) [inline src/client/scenes/mainScene.ts:114]: Empty line

File: src/client/scenes/mainScene.ts [hunk 5]
```diff
@@ -67,45 +69,73 @@ export default class MainScene extends Phaser.Scene implements UpdateListener {
...
+
+      if (cursors.up.isDown && sprite.body.touching.down) {
+        sprite.setVelocityY(-330)
+      }
+
+      if (sprite.x != this.prev_location[0] || sprite.y != this.prev_location[1]) {
+        this.socket.updatePos(sprite.x, sprite.y)
+        this.prev_location = [sprite.x, sprite.y]
+      }
     }
```
Comment (kyleai) [inline src/client/scenes/mainScene.ts:101]: here

File: src/client/scenes/mainScene.ts (hunk 2 above)
Comment (kyleai) [reply src/client/scenes/mainScene.ts:79]: https://github.com/kyleai/socket-io-game/pull/7/files/861ea873ed278ce21644758d92272b60b4a1976e#r436321055

it checks whether we have been given a UUID from the server. if not, we don't want to move

**Pull request merged**